src/hooks/useCalculator.ts

```diff
@@ -0,0 +1,88 @@
+import { useCallback, useState } from 'react';
+
+import { ERROR_MESSAGE, MAX_NUM } from '../constants/calculator';
+
+type Operator = '/' | 'X' | '+' | '-';
+
+export default function useCalculator() {
+  const [firstNum, setFirstNum] = useState('');
+  const [operator, setOperator] = useState<Operator | ''>('');
+  const [view, setView] = useState('0');
+
+  const handleNumberClick = useCallback(
+    (clickedNum: string) => {
+      const newValue =
+        view === '0' || (operator && !firstNum)
+          ? clickedNum
+          : view + clickedNum;
```

Comment:
조건의 복잡도가 높다면 의미있는 변수를 통해 조건을 설명하는 방식을 사용해 볼 수 있습니다.
```suggestion
      const isEmptyView = view === '0'
      const hasInvalidOperation = operator && !firstNum
      const shouldReplaceView = isEmptyView || hasInvalidOperation
      const newValue =shouldReplaceView
          ? clickedNum
          : view + clickedNum;
```